File: app/src/main/java/com/nimtego/plectrum/presentation/main/adapter/MainPagerAdapter.java
package com.nimtego.plectrum.presentation.main.adapter;

import android.content.Context;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

import com.nimtego.plectrum.presentation.main.MainContract;
import com.nimtego.plectrum.presentation.main.fragments.MainTabsFragment;

import java.util.ArrayList;
import java.util.List;

public class MainPagerAdapter extends FragmentPagerAdapter {
    private final List<MainTabsFragment> mFragmentList;
    private final List<String> mFragmentTitleList;

    private MainContract.View parent;

    public MainPagerAdapter(MainContract.View parent, FragmentManager manager) {
        super(manager);
        mFragmentList = new ArrayList<>();
        mFragmentTitleList = new ArrayList<>();
        this.parent = parent;
    }

    @Override
    public MainTabsFragment getItem(int position) {
        return mFragmentList.get(position);
    }

    @Override
    public int getCount() {
        return mFragmentList.size();
    }

    public void addFragment(MainTabsFragment fragment, String title) {
        fragment.onAttach((Context) parent);
        mFragmentList.add(fragment);
        mFragmentTitleList.add(title);
    }

    @Override
    public CharSequence getPageTitle(int position) {
        return mFragmentTitleList.get(position);
    }

}

File: app/src/main/java/com/nimtego/plectrum/presentation/main/adapter/MainPagerAdapter.kt
package com.nimtego.plectrum.presentation.main.adapter

import android.content.Context
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentPagerAdapter
import com.nimtego.plectrum.presentation.main.fragments.MainTabsFragment
import com.nimtego.plectrum.presentation.mvp.MainView
import java.util.*

class MainPagerAdapter(private val parent: MainView, manager: FragmentManager) : FragmentPagerAdapter(manager) {
    private val mFragmentList: MutableList<MainTabsFragment>
    private val mFragmentTitleList: MutableList<String>

    init {
        mFragmentList = ArrayList()
        mFragmentTitleList = ArrayList()
    }

    override fun getItem(position: Int): MainTabsFragment {
        return mFragmentList[position]
    }

    override fun getCount(): Int {
        return mFragmentList.size
    }

    fun addFragment(fragment: MainTabsFragment, title: String) {
        fragment.onAttach(parent as Context)
        mFragmentList.add(fragment)
        mFragmentTitleList.add(title)
    }

    override fun getPageTitle(position: Int): CharSequence {
        return mFragmentTitleList[position]
    }

}